gameplay/gameplay.go

```diff
@@ -0,0 +1,98 @@
+package gameplay
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"reflect"
+	"regexp"
+	"time"
+)
+
+var guesses int
+
+func GetDifficulty() int {
+	var difficulty int
+
+	for {
+		fmt.Print("Please select a difficulty: ")
+		fmt.Print("Easy: 1 ")
+		fmt.Print("Medium: 2 ")
+		fmt.Print("Hard: 3 ")
+		fmt.Scan(&difficulty)
+		if validatedDifficulty(difficulty) {
+			return setGuesses(difficulty)
+		} else {
+			fmt.Println("Sorry, it must be a number between 1 and 3")
+
+		}
+	}
+
+	return difficulty
+}
+
+func validatedDifficulty(d int) bool {
+
+	//check to see if input is type int
+	//used Sprint to cast interface to string so I can compare
+	if fmt.Sprint(reflect.TypeOf(d)) == "int" && d <= 3 {
```

Comment:
This check for `int` is unnecessary since you're passing in an `int` as the parameter. Also this not the correct way to check a type with the reflect package